sdk/core/azure-core/test/e2e/azure_core_with_curl_bodyBuffer.cpp

```diff
@@ -69,7 +71,32 @@ int main()
   return 0;
 }
 
-#ifdef POSIX
+#ifdef _WIN32
+void doFileRequest(Context const& context, HttpPipeline& pipeline)
+{
+  (void)pipeline;
+  Azure::Core::Http::Url host("https://httpbin.org/put");
+  cout << "Creating a File request to" << endl << "Host: " << host.GetAbsoluteUrl() << endl;
+
+  // NOTE: To run the sample: Create folder 'home' on main hard drive (like C:/) and then add a file
+  // `a` in there
+  //
+  HANDLE hFile = CreateFile(
+      "/home/a",
+      GENERIC_READ,
+      FILE_SHARE_READ,
+      NULL,
+      OPEN_EXISTING,
+      FILE_FLAG_SEQUENTIAL_SCAN,
+      NULL);
+  auto requestBodyStream = std::make_unique<FileBodyStream>(hFile, 20, 200);
```

Comment:
We need to check the output of this function call.  If it fails we need to throw,  Failure is denoted by the return code of INVALID_HANDLE_VALUE

Reply:
Again, hhis is not a goal of this PR, I am just moving things around, let's take care of this in a separate PR. @vhvb1989 will you look into this, should be a simple fix, or should I create a work item?

Reply:
Issue for another PR. 
https://github.com/Azure/azure-sdk-for-cpp/issues/1074

Agree with Anton, this is not related for this fix

Reply:
I will close the created issue as not to be fixed since this is only tests which are not intented to validate the open file API.
FileBodyStream takes a file which was already opened and throws if read operation failed.